### ISSUE TITLE
internal error in test-code-generator/FuzzAssembleMove runned on hifive-unleashed failed

### ISSUE THREAD

**Issue by luyahan** [v8-riscv/v8#61]
**Describe the bug**
Run  test-code-generator/FuzzAssembleMove on hifive-unleashed failed, but  Test pass on qemu.

**To Reproduce**
1. Build with options `riscv64.native.debug`
2. Run test/program `./cctest test-code-generator/FuzzAssembleMove --random-seed=658291538 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner`
4. See ` Fatal error`

**Expected behavior**
Run test success.

**Errors/Logs**
```
#
# Fatal error in ../../test/cctest/compiler/test-code-generator.cc, line 796
# Expected: '0.423357 (kRepFloat32 stack slot)' but got 'nan (kRepFloat32 stack slot)'
#
#
#
#FailureMessage Object: 0x3fffbf7b78
==== C stack trace ===============================

    ./cctest(+0x31c2a4c) [0x2ad3254a4c]
    ./cctest(+0x313564c) [0x2ad31c764c]
    ./cctest(+0x2f30692) [0x2ad2fc2692]
    ./cctest(+0x8c199c) [0x2ad095399c]
    ./cctest(+0x8c6b84) [0x2ad0958b84]
    ./cctest(+0x8908f0) [0x2ad09228f0]
    ./cctest(+0x81c4b0) [0x2ad08ae4b0]
    /lib64/lp64d/libc.so.6(__libc_start_main+0xa6) [0x3fd76cd204]
    ./cctest(_start+0x2c) [0x2ad091f894]
Aborted (core dumped)
```

**Additional context**
Board:HIFIVE UNLEASHED
OS:Linux fedora-riscv-sifive 5.5.0-0.rc5.git0.1.1.riscv64.fc32.riscv64 #1 SMP Mon Jan 6 17:31:22 UTC 2020 riscv64 riscv64 riscv64 GNU/Linux

**Comment by luyahan** [comment]
```
(gdb) x /20i 0x3fee582044
   0x3fee582044:        or      a4,a0,zero
   0x3fee582048:        j       0x3fee58205c
   0x3fee58204c:        nop
   0x3fee582050:        nop
   0x3fee582054:        ret
   0x3fee582058:        nop
   0x3fee58205c:        lui     a4,0x0
   0x3fee582060:        addiw   a4,a4,3
   0x3fee582064:        slli    a4,a4,0xc
   0x3fee582068:        addi    a4,a4,-1361
   0x3fee58206c:        slli    a4,a4,0xc
   0x3fee582070:        addi    a4,a4,760
   0x3fee582074:        slli    a4,a4,0xc
   0x3fee582078:        addi    a4,a4,2012
=> 0x3fee58207c:        jr      a4
   0x3fee582080:        nop
   0x3fee582084:        ebreak
   0x3fee582088:        lui     zero,0x54321
   0x3fee58208c:        ebreak
   0x3fee582090:        lui     zero,0x54321
(gdb) info registers a4
a4             0x2aaf2f87dc     183327754204
(gdb) x /20i 0x2aaf2f87dc
   0x2aaf2f87dc:        nop
   0x2aaf2f87e0:        nop
   0x2aaf2f87e4:        ret
   0x2aaf2f87e8:        nop
   0x2aaf2f87ec:        lui     a4,0x0
   0x2aaf2f87f0:        addiw   a4,a4,3
   0x2aaf2f87f4:        slli    a4,a4,0xc
   0x2aaf2f87f8:        addi    a4,a4,-1361
   0x2aaf2f87fc:        slli    a4,a4,0xc
   0x2aaf2f8800:        addi    a4,a4,760
   0x2aaf2f8804:        slli    a4,a4,0xc
   0x2aaf2f8808:        addi    a4,a4,2012
   0x2aaf2f880c:        jr      a4
   0x2aaf2f8810:        nop
   0x2aaf2f8814:        ebreak
   0x2aaf2f8818:        lui     zero,0x54321
   0x2aaf2f881c:        ebreak
   0x2aaf2f8820:        lui     zero,0x54321
   0x2aaf2f8824:        ebreak
   0x2aaf2f8828:        lui     zero,0x54321
```

**Comment by luyahan** [comment]
In the latest code, there is no this problem, to be confirmed in detail next week.

**Comment by lazyparser** [comment]
interesting. keep this issue for one or two weeks to see if it happens again.

**Comment by luyahan** [comment]
The internal error still exit.

**Comment by luyahan** [comment]
I guess it is smilar with #74

**Comment by luyahan** [comment]
fix by #105
